hs/src/Reach/Parser.hs

```diff
@@ -223,50 +276,92 @@ tryPrettifyError at' e = case readMaybe e of
         _ -> ""
   _ -> e
 
-gatherDeps_ast_rewriteErr :: SrcLoc -> IORef JSBundleMap -> String -> IO [JSModuleItem]
-gatherDeps_ast_rewriteErr at' fmr s = case parseModule s (show $ get_srcloc_src at') of
+gatherDeps_ast_rewriteErr
+  :: Maybe HostGit
+  -> SrcLoc
+  -> IORef JSBundleMap
+  -> String
+  -> ReaderT ParserOpts IO [JSModuleItem]
+gatherDeps_ast_rewriteErr mh at' fmr s = case parseModule s (show $ get_srcloc_src at') of
   Left e -> error $ tryPrettifyError at' e -- TODO: prettify
-  Right r -> gatherDeps_ast at' fmr r
+  Right r -> gatherDeps_ast mh at' fmr r
 
 data GatherContext = GatherTop | GatherNotTop
   deriving (Eq)
 
-gatherDeps_file :: GatherContext -> SrcLoc -> IORef JSBundleMap -> FilePath -> IO FilePath
-gatherDeps_file gctxt at fmr src_rel =
-  updatePartialAvoidCycles at fmr (gatherDeps_from at) [ReachStdLib] get_key ret_key err_key proc_key
+gatherDeps_file
+  :: GatherContext
+  -> Maybe HostGit
+  -> SrcLoc
+  -> IORef JSBundleMap
+  -> FilePath
+  -> ReaderT ParserOpts IO FilePath
+gatherDeps_file gctxt mh at fmr src_rel = do
+  updatePartialAvoidCycles
+    at
+    fmr
+    (gatherDeps_from at)
+    [ReachStdLib]
+    get_key
+    ret_key
+    Err_Parse_CyclicImport
+    proc_key
+
   where
-    get_key () = do
-      src_abs <- makeAbsolute src_rel
-      reRel <- makeRelativeToCurrentDirectory src_abs
-      when (gctxt == GatherNotTop) $ do
-        when (isAbsolute src_rel) $ do
-          expect_thrown at (Err_Parse_ImportAbsolute src_rel)
-        when ("../" `isPrefixOf` reRel) $ do
-          expect_thrown at (Err_Parse_ImportDotDot src_rel)
-      return $ ReachSourceFile src_abs
-    ret_key (ReachSourceFile x) = x
-    ret_key (ReachStdLib) = no_stdlib
     no_stdlib = impossible $ "gatherDeps_file: source file became stdlib"
-    err_key x = Err_Parse_CyclicImport x
-    proc_key (ReachStdLib) = no_stdlib
-    proc_key src@(ReachSourceFile src_abs) = do
+
```

Comment:
This code chunk uses `Either` rather than just pushing the `expect_thrown` into the error site, which is our style